packages/hardhat/contracts/strategy/CrabMigration.sol

```diff
@@ -130,35 +174,100 @@ contract CrabMigration is Ownable {
+            uint256 crabV1Balance = crabV1.balanceOf(address(this));
+
+            // 2. mint osqth in crab v2
+            uint256 wSqueethToMint = crabV1.getWsqueethFromCrabAmount(crabV1Balance);
+            uint256 timeAtLastHedge = crabV1.timeAtLastHedge();
+            uint256 priceAtLastHedge = crabV1.priceAtLastHedge();
+            crabV2.initialize{value: _amount}(
+                wSqueethToMint,
+                totalCrabV1SharesMigrated,
+                timeAtLastHedge,
+                priceAtLastHedge
+            );
+
+            // 3. call withdraw from crab v1
+            IERC20(wPowerPerp).approve(address(crabV1), type(uint256).max);
+            crabV1.approve(address(crabV1), crabV1Balance);
+            crabV1.withdraw(crabV1Balance);
+        } else if (FLASH_SOURCE(_callSource) == FLASH_SOURCE.FLASH_MIGRATE_V1_TO_V2) {
+            FlashMigrateV1toV2 memory data = abi.decode(_calldata, (FlashMigrateV1toV2));
+
+            crabV2.deposit{value: _amount}();
+
+            crabV1.transferFrom(_initiator, address(this), data.crabV1ToWithdraw);
+
+            IERC20(wPowerPerp).approve(address(crabV1), data.v1oSqthToPay);
+            crabV1.withdraw(data.crabV1ToWithdraw);
+
+            // Flash deposit remaining ETH, if user said so. Else return back the ETH. If CR1 = CR2 ethToFlashDeposit should be 0
+            if (data.ethToFlashDeposit > 0) {
+                crabV2.flashDeposit{value: address(this).balance - _amount}(data.ethToFlashDeposit);
+            }
+
+            // Sent back the V2 tokens to the user
+            crabV2.transfer(_initiator, crabV2.balanceOf(address(this)));
+            IERC20(wPowerPerp).transfer(_initiator, IERC20(wPowerPerp).balanceOf(address(this)));
+
+            // Sent back the excess ETH
+            if (address(this).balance > _amount) {
+                payable(_initiator).sendValue(address(this).balance - _amount);
+            }
+        } else if (FLASH_SOURCE(_callSource) == FLASH_SOURCE.FLASH_MIGRATE_WITHDRAW_V1_TO_V2) {
+            FlashMigrateAndBuyV1toV2 memory data = abi.decode(_calldata, (FlashMigrateAndBuyV1toV2));
+            (, , , uint256 v1Short) = crabV1.getVaultDetails();
+
+            crabV1.transferFrom(_initiator, address(this), data.crabV1ToWithdraw);
+            crabV2.deposit{value: _amount}();
+
+            uint256 oSqthToPay = IERC20(wPowerPerp).balanceOf(address(this));
+            IERC20(wPowerPerp).approve(address(crabV1), oSqthToPay);
+
+            // Find crab amount for contract's sqth balance. Remaining crab sould be withdrawn using flash withdraw
+            uint256 crabV1ToWithdrawRmul = oSqthToPay.wmul(crabV1.totalSupply()).rdiv(v1Short);
+            uint256 crabV1ToWithdraw = crabV1ToWithdrawRmul.floor(10**9) / (10**9);
+
+            crabV1.withdraw(crabV1ToWithdraw);
+
+            crabV1.flashWithdraw(data.crabV1ToWithdraw - crabV1ToWithdraw, data.withdrawMaxEthToPay);
```

Comment:
only if(data.crabV1ToWithdraw - crabV1ToWithdraw >0)

Reply:
I think it's It should be greater than 0!